tendenci/apps/events/forms.py

```diff
@@ -965,15 +965,20 @@ def __init__(self, *args, **kwargs):
             #self.fields['groups'].initial = Group.objects.get_or_create_default()
             
         if 'repeat_of' in self.fields:
-            if self.instance.pk and self.instance.parent: # on event edit
-                if self.instance.repeat_of: # repeat_of already set
-                    self.fields['repeat_of'].queryset = Event.objects.filter(
-                        pk=self.instance.repeat_of.pk)
-                else:
-                    self.fields['repeat_of'].queryset = Event.objects.filter(parent_id=self.instance.parent.id).order_by('repeat_uuid').distinct('repeat_uuid')
-            elif 'parent' in self.fields and parent_event_id: # on sub-event add
-                parent_event = Event.objects.get(id=parent_event_id)
-                self.fields['repeat_of'].queryset = parent_event.child_events.order_by('repeat_uuid').distinct('repeat_uuid')
+            if self.instance.pk and self.instance.parent and self.instance.repeat_of: # on event edit, repeat_of already set
+                self.fields['repeat_of'].queryset = Event.objects.filter(pk=self.instance.repeat_of.pk)
+            elif (self.instance.pk and self.instance.parent) or ('parent' in self.fields and parent_event_id): # on editing sub-event with no repeat, or adding new sub-event
```

Comment:
If there's an instnace and it has a parent & repeat_of is set, then limit drop down to the repeat_of it's set to